### PR TITLE
turn cookie store detection on it's feet

### PR DESCRIPTION
**Pull request by moench-tegeder**
All the Unix-like platforms (Linux, *BSD) use the same location for
the browser profile directories, so it makes some sense to make
that the default instead of tracking all the platform names; and
treat Windows and OSX as the exceptions they are (based on the number
of platform names, not installs). This is especially useful as the
"platform" name on at least FreeBSD changes with each major release
(e.g. freebsd12, freebsd13, ...) and would have to be updated for
every release without this change here.
At the same time, fix the error condition: the question is not whether
we are on a "supported" platform but if the browser profile directory
exists in the place where we expect it. That handles all the cases like
"browser not available", "browser not installed", "profile not
initialized (yet)" and "non-standard profile dir location". Change
error message to match that behaviour.

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible.

### CODE REVIEW

File: yt_dlp/cookies.py [hunk 1]
```diff
@@ -156,30 +156,22 @@ def _extract_firefox_cookies(profile, logger):
...
-        return os.path.expanduser('~/Library/Application Support/Firefox')
+        profpath = os.path.expanduser('~/Library/Application Support/Firefox')
     else:
-        raise ValueError(f'unsupported platform: {sys.platform}')
+        profpath = os.path.expanduser('~/.mozilla/firefox')
+
+    if(os.path.exists(profpath)):
+        return profpath
+    else:
+        raise ValueError(f'cannot find profiles dir (unsupported platform: {sys.platform}?)')
```
Comment (pukkandan) [inline yt_dlp/cookies.py:169]: 1. `_extract_chrome_cookies` etc checks whether the folder is found. It doesn't need to be checked here`
2. Windows can be `cygwin` as well

```suggestion
    if sys.platform in ('win32', 'cygwin'):
        return os.path.expandvars(R'%APPDATA%\Mozilla\Firefox\Profiles')
    elif sys.platform == 'darwin':
        return os.path.expanduser('~/Library/Application Support/Firefox')
    return os.path.expanduser('~/.mozilla/firefox')
```

File: yt_dlp/cookies.py [hunk 2]
```diff
@@ -217,6 +217,9 @@ def _get_chromium_based_browser_settings(browser_name):
     }[browser_name]
 
     browsers_without_profiles = {'opera'}
+    if browser_name not in browsers_without_profiles:
+        if not os.path.exists(browser_dir):
+            raise ValueError(f'browser profile dir not found (unsupported platform: {sys.platform}?)')
```
Comment (pukkandan) [inline yt_dlp/cookies.py:222]: see above
```suggestion
```

File: yt_dlp/cookies.py [hunk 3]
```diff
@@ -156,30 +156,22 @@ def _extract_firefox_cookies(profile, logger):
...
-            'brave': os.path.join(config, 'BraveSoftware/Brave-Browser'),
-            'chrome': os.path.join(config, 'google-chrome'),
-            'chromium': os.path.join(config, 'chromium'),
-            'edge': os.path.join(config, 'microsoft-edge'),
-            'opera': os.path.join(config, 'opera'),
-            'vivaldi': os.path.join(config, 'vivaldi'),
-        }[browser_name]
-
-    elif sys.platform == 'win32':
+    if sys.platform == 'win32':
```
Comment (pukkandan) [inline yt_dlp/cookies.py:174]:
```suggestion
    if sys.platform in ('win32', 'cygwin'):
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Comment by pukkandan** [comment]
Replaces https://github.com/yt-dlp/yt-dlp/pull/3370

File: yt_dlp/cookies.py (hunk 1 above)
Comment (moench-tegeder) [reply yt_dlp/cookies.py]: OK, Im always thinking "fail early", but let's not open that can of worms here.

File: yt_dlp/cookies.py [hunk 4]
```diff
@@ -156,30 +156,18 @@ def _extract_firefox_cookies(profile, logger):
...
-    elif sys.platform == 'win32':
+    if sys.platform in ('cygwin', 'win32'):
         return os.path.expandvars(R'%APPDATA%\Mozilla\Firefox\Profiles')
     elif sys.platform == 'darwin':
         return os.path.expanduser('~/Library/Application Support/Firefox')
-    else:
-        raise ValueError(f'unsupported platform: {sys.platform}')
+
+    # fallthrough case: all the other Linux/BSD/Unix derivatives
+    return os.path.expanduser('~/.mozilla/firefox')
```
Comment (pukkandan) [inline yt_dlp/cookies.py:165]:
```suggestion
    return os.path.expanduser('~/.mozilla/firefox')
```

File: yt_dlp/cookies.py [hunk 5]
```diff
@@ -202,8 +190,17 @@ def _get_chromium_based_browser_settings(browser_name):
             'vivaldi': os.path.join(appdata, 'Vivaldi'),
         }[browser_name]
 
+    # all the other Linux/BSD/Unix derivatives
```
Comment (pukkandan) [inline yt_dlp/cookies.py:193]:
```suggestion
```

**Pull request merged**